### PR TITLE
Remove unused part I

### PR DESCRIPTION
**Pull request by jhprinz**
Removes unused and legacy parts we agreed upon. See #566

### CODE REVIEW

**Comment by jhprinz** [comment]
I have done all the things besides test cleanup in #566. I added a message to the `iter_extendable_slices` function that it is experimental and not properly tested. We can do that later, if we want. Otherwise this is ready to be merged when it passes.

**Comment by jhprinz** [comment]
great. Ready to merge. I removes lot of stuff and by that increases coverage. I needed to change some copy to replace by equivalent functions.

File: openpathsampling/engines/trajectory.py [hunk 1]
```diff
@@ -99,18 +99,6 @@ def reversed(self):
...
-        """
-        Prepend a snapshot
-
-        Just convenience method to replace insert(0, snapshot)
-        """
-        self.insert(0, snapshot)
-        # And a generation of scientist-programmers who grew up learning
-        # "OPS trajectories are just Python lists" scream in pain when they
-        # find this after googling "python list.prepend not working".
-        # (Blame JHP. This was his doing.) ;)
```
Comment (dwhswenson) [inline openpathsampling/engines/trajectory.py]: Alas, the end of one of my favorite comments in our code.

Still, for the better.

File: openpathsampling/sample.py [hunk 2]
```diff
@@ -119,14 +121,15 @@ def __setitem__(self, key, value):
...
     def __eq__(self, other):
-        if len(self.samples) == len(other.samples):
-            for samp1 in self.samples:
-                if samp1 not in other.samples:
-                    return False
-
-            return True
-        else:
-            return False
+        return set(self.samples) == set(other.samples)
```
Comment (dwhswenson) [inline openpathsampling/sample.py]: can you add a length check to this as well? you could have duplicated samples in a sample set, then I would argue that these wouldn't be equal

**Comment by dwhswenson** [comment]
Could you clear outputs/check out clean versions of AD TPS notebooks? I'd like the stuff I'm running now to have a chance at being the last versions. Unless there's new code from yours that is actually necessary (most of it looks like temporary debug that isn't needed for example purposes)

**Comment by dwhswenson** [comment]
A couple small suggestions, otherwise this looks good! Nice to bump up 0.5% in coverage for the code equivalent of picking things up from the living room floor. :-)

File: openpathsampling/engines/trajectory.py (hunk 1 above)
Comment (jhprinz) [inline openpathsampling/engines/trajectory.py]: I might add some other place for you in the future :)

File: openpathsampling/sample.py (hunk 2 above)
Comment (jhprinz) [inline openpathsampling/sample.py]: Important, I checked the code again and it seems that we agreed before that one sample can only be present once. If you apply samples then the one with the same replica number will be replaced. So, at least by applying you cannot have the two samples with the same replica number present. Just to make sure I will change it to using a `collections.Counter` which will also compare the numbers of times a specific sample is present.

**Comment by jhprinz** [comment]
If it passes, it is ready.

**Comment by jhprinz** [comment]
> Nice to bump up 0.5% in coverage

Yes, I like this way of writing tests...

File: openpathsampling/sample.py (hunk 2 above)
Comment (dwhswenson) [inline openpathsampling/sample.py]: `collections.Counter` is even better than a length check. My concern is that there are also `append` and `extend` methods for `SampleSet`, which could create a `SampleSet` where you'd get false equality from the set-based version. (Didn't there used to be an `apply` method which did the cleanup to make sure there was only one of each? There should be such a thing, but I don't see it in the code now.)

**Comment by dwhswenson** [comment]
Merging -- please note my question in the [line note above](https://github.com/openpathsampling/openpathsampling/pull/572#discussion_r77439248). That's not to be done in this PR, but we might need to add that behavior.

**Pull request merged**